libs/community/langchain_community/cache.py

```diff
@@ -350,20 +350,24 @@ def clear(self, **kwargs: Any) -> None:
 
 
 class RedisCache(BaseCache):
-    """Cache that uses Redis as a backend."""
+    """
+    Cache that uses Redis as a backend. Allows to use either sync or
+    async Redis client. Depending on the client passed, you are expected
+    to use either sync or async methods.
+    """
 
     def __init__(self, redis_: Any, *, ttl: Optional[int] = None):
```

Comment:
Maybe for this you could have a `RedisCache` that uses a `redis.Redis` client and an `AsyncRedisCache` that uses a `redis.asyncio.Redis` ? (I know this is the opposite move as for BaseCache 😄 )

Reply:
I actually did it this way in the very first variant, but I didn't like it was literally 99% of the copy from the original RedisCache. The remaining 1% were async/await keywords and client check in constructor. Feels like a potential future maintenance problem?

Reply:
Also, I am not sure what I suppose to do for `AsyncRedisCache` sync methods `lookup/update/clear`. Like I can raise error on those, or do the opposite trick for async in sync execution, but doesn't it feel like too much of duplication of effort?

Reply:
You could put this common behavior in a parent abstract class ?

Reply:
Alright, I can do it. But before that, can you tell me whats your rationale here? Just so I know for future :)

Reply:
@cbornet I mean I was asking whats the motivation for such split. 

I've pushed the implementation which is basically what you said, however I went a bit further with sync-in-async and actually made it so you can run it in certain context (weird case though, I log warning for that). I can rework to `NotImplementedError` if you think its cleaner.

Reply:
> or do the opposite trick for async in sync execution
 
At the moment, we generally try to avoid that. Only one event loop is allowed in a given thread, so there's some additional logic that's required to determine if there's an event loop running already and if yes, kick off another thread etc